src/gpu/generic/sycl/sycl_primitive_conf.hpp

```diff
@@ -415,6 +415,39 @@ struct sycl_pooling_bwd_conf_t : public sycl_pooling_base_conf_t {
     xpu::sycl::md_t diff_dst_md;
 };
 
+struct sycl_reduction_conf_t {
+    xpu::sycl::md_t src_md;
+    xpu::sycl::md_t dst_md;
+    dnnl_alg_kind_t alg = dnnl_alg_kind_undef;
+    float p = 0;
+    float eps = 0;
+    data_type_t src_dt;
+    data_type_t dst_dt;
+    data_type_t local_mem_dt = dnnl_f32;
+
+    int num_dims = 0;
+    xpu::sycl::md_t::dim32_t src_dims[xpu::sycl::md_t::max_dims] = {};
+    int num_axes = 0;
+    xpu::sycl::md_t::dim32_t axes[xpu::sycl::md_t::max_dims] = {};
+    int batch_size = 1;
+    int reduce_size = 1;
+    int stride_size = 1;
+    int tile_row;
+    int tile_col;
+    int batch_groups = -1;
+
+    bool transpose;
+    bool bank_offset;
+    bool is_first_iter;
+    bool is_last_iter;
+
+    sycl_post_ops_t post_ops;
+
+    const int num_sg_reductions = 4;
+    static constexpr int local_row_wg = 256;
+    static constexpr int local_col_wg = 8;
```

Comment:
Any reason for these values (maybe adding a few comments to clarify)?

Reply:
These were derived from testing and are values that should work in most cases. I added a comment to actually tune them in the future when we do an optimisation pass.